src/Expr.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+/**
+ * A representation of a FaunaDB Query Expression. Generally, you shouldn't need
+ * to use this class directly; use the Query helpers defined in {@link module:query}.
+ *
+ * @param {Object} obj The object that represents a Query to be treated as an Expression.
+ * @constructor
+ */
+function Expr(obj) {
+  this.raw = obj;
+}
+
+/**
+ * Wraps an object as an Expression. This will automatically wrap any bare objects with
+ * the appropriate {@link object} escaping.
+ * @param {Object} obj
+ *  The object to be wrapped as an Expression.
+ * @returns {Expr}
+ *   The expression wrapping the provided object.
+ */
+Expr.wrap = function(obj) {
```

Comment:
and removing the special handling of lambda args in the query expression constructors.